### PR TITLE
adding solrcloud pack

### PR DESCRIPTION
**Pull request by swathisingamsetty12**

### CODE REVIEW

**Comment by gauravlall** [comment]
@swathisingamsetty12  Please remove  .DS_Store and walmart specific info like nexus will review after these are removed.

File: components/cookbooks/solr-monitor/metadata.rb [hunk 1]
```diff
@@ -0,0 +1,122 @@
+name             'Solr-monitor'
+maintainer       'forklift search'
+maintainer_email 'ssingamsetty@walmartlabs.com'
+license          'All rights reserved'
```
Comment (schwankl) [inline components/cookbooks/solr-monitor/metadata.rb]: change to apache 2

File: components/cookbooks/solr-monitor/recipes/deploysolrmonitoring.rb [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+      rm -rf crontab-#{solr_monitor_version}.txt
+      rm -rf #{node['monitor']['dir']}/solr_monitor_crontab.txt
+    EOH
+  end
+end
+
+bash 'createcrontabscript' do 
+  code <<-EOH
+    cd "#{node['user']['dir']}"
+    echo '*/1 * * * * #{node['monitor']['dir']}/solr-monitor/solr_monitor.py -solrcollection #{logical_collection_name} -solrhost http://#{node['ipaddress']}:8080/ -graphiteserver #{graphite_server} -id #{node['ipaddress']} -graphitecollname #{logical_collection_name} -statsprefix #{app_name} -graphiteport #{graphite_port} -defaultemail #{email_addresses} -datacenter #{solrcloud_datacenter} -env #{solrcloud_env}' >> #{node['monitor']['dir']}/solr_monitor_crontab.txt
```
Comment (schwankl) [inline components/cookbooks/solr-monitor/recipes/deploysolrmonitoring.rb]: would be cleaner, easier to maintain if you use the chef cron resource

Comment (swathisingamsetty12) [inline components/cookbooks/solr-monitor/recipes/deploysolrmonitoring.rb]: the chef cron resource does not shedules for every 'x' no of minutes . And also we need to reset the cron job whenever the ip addresses change in the solrcloud .
ex: chef cron resource gives ( 5 * * * * * command ) and i need ( */5 * * * * command ).

File: components/cookbooks/solr-monitor/metadata.rb (hunk 1 above)
Comment (swathisingamsetty12) [inline components/cookbooks/solr-monitor/metadata.rb]: I have removed the email address . Can the maintainer be "forklift search" .

File: components/cookbooks/solr-monitor/recipes/deploysolrmonitoring.rb (hunk 2 above)
Comment (schwankl) [inline components/cookbooks/solr-monitor/recipes/deploysolrmonitoring.rb]: chef cron resource supports it - example http://stackoverflow.com/questions/28441284/schedule-cron-every-3-hour-in-chef-recipe

Comment (swathisingamsetty12) [inline components/cookbooks/solr-monitor/recipes/deploysolrmonitoring.rb]: solr-monitor component depends on internal walmart specific url and not doing open source now. I will make this change internally .

File: components/cookbooks/solr-monitor/metadata.rb (hunk 1 above)
Comment (swathisingamsetty12) [inline components/cookbooks/solr-monitor/metadata.rb]: updated

File: components/cookbooks/solrcloud/metadata.rb [hunk 3]
```diff
@@ -0,0 +1,213 @@
+name 'Solrcloud'
+license 'none'
```
Comment (schwankl) [inline components/cookbooks/solrcloud/metadata.rb]: should be "Apache License, Version 2.0" like all the others

Comment (swathisingamsetty12) [inline components/cookbooks/solrcloud/metadata.rb]: added licence version

File: components/cookbooks/solrcloud/attributes/default.rb [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+default['protocol'] = "#{node['protocol']}";
+
+default['tomcatversion'] = "#{node['tomcat_version']}"[0,1];
+default['tomcat']['dir'] = "#{node['user']['dir']}/tomcat#{node['tomcatversion']}";
+
+default['clusterstatus']['uri'] = "solr/admin/collections?action=CLUSTERSTATUS&wt=json";
+
+default['log4j']['version'] = "1.2.17";
+default['log4j']['name'] = "log4j-#{node['log4j']['version']}.jar";
+default['log4j']['url'] = "http://gec-maven-nexus.walmart.com/nexus/content/groups/public/log4j/log4j/#{node['log4j']['version']}/#{node['log4j']['name']}";
```
Comment (schwankl) [inline components/cookbooks/solrcloud/attributes/default.rb]: no walmart.com specific stuff in open source / circuit-oneops-1

File: components/cookbooks/solrcloud/libraries/java_util.rb [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+
+module Java
+
+  module Util
+
+    require 'json'
+
+    include Chef::Mixin::ShellOut
+
+    # def getmirrorservice
```
Comment (schwankl) [inline components/cookbooks/solrcloud/libraries/java_util.rb]: why these 2 block comments?  if not used rm

Comment (swathisingamsetty12) [inline components/cookbooks/solrcloud/libraries/java_util.rb]: not using those 2 methods . will remove them

File: components/cookbooks/solrcloud/attributes/default.rb (hunk 4 above)
Comment (swathisingamsetty12) [inline components/cookbooks/solrcloud/attributes/default.rb]: removed wamart specific urls

**Comment by swathisingamsetty12** [comment]
closing this PR and raised https://github.com/oneops/circuit-oneops-1/pull/124 to 3-31-2016 branch.

**Pull request closed without merging**